### ISSUE TITLE
[Feat] Add support to other color representations YUV

### ISSUE THREAD

**Issue by edgarriba** [kornia/kornia#934]
## 🚀 Feature
 Add support to other color representations YUV 420, 444

**Comment by edgarriba** [comment]
Do you have any reference material for what you describe ? This was an external feature request, after quick search I found the following snippet.
https://gist.github.com/Quasimondo/c3590226c924a06b276d606f4f189639

**Comment by cceyda** [comment]
> 3. [B,1,H,W] (Y) + [B,2,H/2,W/2] for UV (or UV split as well into 3 different tensors). Somewhat sensible but having multiple tensors breaks with other formats.

This option looks most reasonable.  Looks like opencv also supports some other YUV formats and has a function to;
"Converts an image from one color space to another **where the source image is stored in two planes**."
https://docs.opencv.org/4.5.3/d8/d01/group__imgproc__color__conversions.html#ga8e873314e72a1a6c0252375538fbf753

Would something like that work?

**Comment by oskarflordal** [comment]
@edgarriba: Right, that link describes the color conversion itself which you already support (at least in a limited fashion, there are multiple different ways of doing this, i.e. rec 709, 2020 as well as 601). Problems comes form doing subsampling as in the case of YUV 420 where we end up with chroma channels that are lower resolution. 420 is 2x reduction both horizontal vertical while 422 split only horizontal. https://en.wikipedia.org/wiki/Chroma_subsampling is a good start.
@cceyda: Agreed that probably does make most  sense. I guess we can limit to just 2 planes as in that example and perhaps do 420 and 422 with just one chrome offset mode (i.e. defining were the chroma samples reside in relation to the luma samples)

**Comment by edgarriba** [comment]
if understand correctly we are targeting the following:
```python
def rgb_to_yub(img: Tensor) -> Tuple[Tensor, Tensor]:
    """Converts a tensor image from RGB to YUV color space.

    Args:
       img: the input RGB image with shape (*3,H,W).

    Returns:
       A tuple containing the converted YUV image.
       - Tensor containing the Y plane with shape (*,1,H,W)
       - Tensor containing the UV plane with shape (*,2,H,W)
    """
    ....
```
The only concern is that later to compose functions will make usability a bit complicated
```python
pipe = nn.Sequential(
    RgbToYuv(),
    RandomRotate(),  # this expects consistent tensor image
)
out = pipe(torch.rand(1,3,224,224)
```

**Comment by edgarriba** [comment]
@oskarflordal that's awesome - definitely we can iterate over that.
